api/controllers/app.api.comments.controllers.js

```diff
@@ -0,0 +1,16 @@
+const { checkArticleExists } = require("../models/app.api.articles.models");
+const {
+  fetchCommentsByArticleId,
+} = require("../models/app.api.comments.models");
+
+exports.getCommentsByArticleId = (req, res, next) => {
+  const { article_id } = req.params;
+
+  checkArticleExists(article_id)
+    .then(() => {
+      fetchCommentsByArticleId(article_id).then((commentsForArticleId) => {
```

Comment:
This is nested. Make sure you change this, as you can't have a `.then()` block within another `.then()` block.